src/frontend/src/common/components/library/data_subview/components/CreateNSTreeModal/index.tsx

```diff
@@ -193,10 +193,13 @@ export const CreateNSTreeModal = ({
         </NextLink>
       </Notification>
       <StyledDialog
-        disableBackdropClick
         disableEscapeKeyDown
         open={open}
-        onClose={handleClose}
+        onClose={(_event, reason) => {
```

Comment:
I don't love having to repeat the same boilerplate for every `onClose` prop. Especially if this version ever gets deprecated as well. Maybe we could instead create a helper func we stash in some shared util and they can all import it? Something like
```
export function noBackdropCloseHandler(handleClose: () => void) {
  return (_event, reason) => {
    if (reason !== "backdropClick") {
      handleClose();
    }
  };
}
```

Then it would get used as just
```
<Dialog
  ...other props ...
  onClose={noBackdropCloseHandler(handleClose)}
  ...
/>
```

Don't get me wrong, this doesn't feel super elegant either, but at least it abstracts it away to a shared location in code. Also not sure how good a name `noBackdropCloseHandler` is, but I'm straining to come up with a better name that takes no more than five words...

My real preference would be to have our own Dialog component that just handled this for us, but since we build off MUI and each of our dialogs are kinda different, I don't think that's going to be a slam dunk either...

Reply:
i think we could create our own dialog component and that would work.. one sec